dask/array/reductions.py

```diff
@@ -82,26 +165,28 @@ def _tree_reduce(x, aggregate, axis, keepdims, dtype, split_every=None,
         n = builtins.max(int(split_every ** (1 / (len(axis) or 1))), 2)
         split_every = dict.fromkeys(axis, n)
     else:
-        split_every = dict((k, v) for (k, v) in enumerate(x.numblocks) if k in axis)
+        raise ValueError("split_every must be a int or a dict")
```

Comment:
Removed undocumented _feature_ where ``split_every="your mom"`` meant a reduction in exactly 2 passes. We could formally reintroduce it in the form of ``split_every=-1`` as part of a separate PR.

Reply:
Also, split_every=None defaults to ``config.get('split_every', 4)``, while split_every=dict() defaults to 2 for the missing axes. Does not seem very coherent to me - but changing it would be out of scope of this PR.